pyth_observer/check/publisher.py

```diff
@@ -134,11 +138,25 @@ def __init__(self, state: PublisherState, config: PublisherCheckConfig):
         self.__state = state
         self.__max_slot_distance: int = int(config["max_slot_distance"])
         self.__abandoned_slot_distance: int = int(config["abandoned_slot_distance"])
+        self.__market_open: bool = is_market_open(
+            self.__state.asset_type.lower(),
+            datetime.now(ZoneInfo("America/New_York")),
+        )
```

Comment:
initialize it here and introduced a new `market_open()` function to retrieve the state so that it can be used in `check_publisher()`